sale_commission/models/account_move.py

```diff
@@ -234,3 +251,57 @@ def _skip_future_payments(self, date_payment_to):
             if any(date_payment_to < date for date in payments_dates):
                 return True
         return False
+
+    def _partial_commissions(self):
+        partial_settlements = self.env["account.invoice.line.agent"]
+        lines_to_update = {}
+        partial_payment_remaining = {}
+        for line in self:
+            line_total_amount = line.amount
+            for (
+                partial,
+                amount,
+                _counterpart_line,
+            ) in line.invoice_id._get_reconciled_invoices_partials():
+                if partial.partial_commission_settled:
+                    continue
+                if partial.id in partial_payment_remaining:
+                    payment_amount = partial_payment_remaining[partial.id][
+                        "remaining_amount"
+                    ]
+                else:
+                    payment_amount = amount
+                    partial_payment_remaining[partial.id] = {"remaining_amount": amount}
+                if line.object_id.price_total < payment_amount:
+                    lines_to_update[line.id] = {
+                        "amount": line_total_amount,
+                        "is_partial_settled": True,
+                    }
+                    partial_payment_remaining[partial.id] = {
+                        "remaining_amount": amount - line.object_id.price_total
+                    }
+                    break
+                paid_in_proportion = payment_amount / line.invoice_id.amount_total
+                partial_commission = (
+                    line.invoice_id.commission_total * paid_in_proportion
+                )
+                if line.id not in lines_to_update:
+                    lines_to_update[line.id] = {
+                        "amount": line_total_amount - partial_commission
+                    }
+                elif partial_commission >= lines_to_update[line.id]["amount"]:
+                    lines_to_update[line.id] = {
+                        "amount": lines_to_update[line.id]["amount"],
+                        "is_partial_settled": True,
+                    }
```

Comment:
**question**: Here you want to update the existing dictionary, is that correct?

if so, wouldn't it be better to simply write "is_partial_settled"?

e.g. `lines_to_update[line.id].update({"is_partial_settled": True})` or `lines_to_update[line.id]["is_partial_settled"] = True)`